### PR TITLE
[RFR] Convert DataGrid component to useStyles

### PR DESCRIPTION
**Pull request by jaytula**
Convert DataGrid component to useStyles #3453

### CODE REVIEW

File: packages/ra-ui-materialui/src/list/Datagrid.js [hunk 1]
```diff
@@ -100,14 +94,40 @@ const styles = theme =>
...
+        setSort,
+        total,
+        version,
+        ...rest
+    } = props;
+
+    const updateSort = event => {
         event.stopPropagation();
-        this.props.setSort(event.currentTarget.dataset.sort);
+        props.setSort(event.currentTarget.dataset.sort);
```
Comment (djhi) [inline packages/ra-ui-materialui/src/list/Datagrid.js]: why use `props`, `setSort` is extracted at L118?

File: packages/ra-ui-materialui/src/list/Datagrid.js [hunk 2]
```diff
@@ -100,14 +94,40 @@ const styles = theme =>
...
     };
 
-    handleSelectAll = event => {
-        const { onSelect, ids, selectedIds } = this.props;
+    const handleSelectAll = event => {
+        const { onSelect, ids, selectedIds } = props;
```
Comment (djhi) [inline packages/ra-ui-materialui/src/list/Datagrid.js]: Same

File: packages/ra-ui-materialui/src/list/Datagrid.js [hunk 3]
```diff
@@ -100,14 +94,40 @@ const styles = theme =>
...
+        rowClick,
+        rowStyle,
+        selectedIds,
+        setSort,
+        total,
+        version,
+        ...rest
+    } = props;
+
+    const updateSort = event => {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/Datagrid.js]: please use `useCallback` for this handler and the others

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**